front/hooks/useFileUploaderService.ts

```diff
@@ -0,0 +1,203 @@
+import type { SupportedContentFragmentType } from "@dust-tt/types";
+import { isSupportedTextContentFragmentType } from "@dust-tt/types";
+import { useContext, useState } from "react";
+
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import { extractTextFromPDF } from "@app/lib/client/handle_file_upload";
+import { getMimeTypeFromFile } from "@app/lib/file";
+
+interface FileContent {
+  content: string;
+  file: File;
+  filename: string;
+  id: string;
+  preview?: string;
+  size: number;
+  // TODO(2024-06-26 flav) Make this configurable.
+  contentType: SupportedContentFragmentType;
+}
+
+type FileBlobUploadErrorCode =
+  | "failed_to_read_file"
+  | "file_too_large"
+  | "file_type_not_supported"
+  | "combined_size_exceeded";
+
+class FileBlobUploadError extends Error {
+  constructor(
+    readonly code: FileBlobUploadErrorCode,
+    readonly file: File,
+    msg?: string
+  ) {
+    super(msg);
+  }
+}
+
+const COMBINED_MAX_FILES_SIZE = 500_000;
+const MAX_FILE_SIZE = 100_000_000;
+
+export function useFileUploaderService() {
+  const [fileBlobs, setFileBlobs] = useState<FileContent[]>([]);
+  const [isProcessingFiles, setIsProcessingFiles] = useState(false);
+
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const handleFileChange = async (e: React.ChangeEvent) => {
+    const selectedFiles = Array.from(
+      (e?.target as HTMLInputElement).files ?? []
+    );
+
+    setIsProcessingFiles(true);
+
+    // TODO(2024-06-26 flav): Handle image's sizes differently.
+    const totalSize = [...fileBlobs, ...selectedFiles].reduce(
+      (sum, content) => sum + content.size,
+      0
+    );
+    if (totalSize > COMBINED_MAX_FILES_SIZE) {
+      sendNotification({
+        type: "error",
+        title: "Files too large.",
+        description:
+          "The combined extracted text from the files you selected results in more than 500,000 characters. This will overflow the assistant context. Please consider uploading smaller files.",
+      });
+      return;
+    }
+
+    try {
+      const previewPromises: Promise<FileContent>[] = selectedFiles.map(
+        async (file) => {
+          const contentType = getMimeTypeFromFile(file);
+          if (!isSupportedTextContentFragmentType(contentType)) {
+            return Promise.reject(
+              new FileBlobUploadError("file_type_not_supported", file)
+            );
+          }
+
+          if (file.size > MAX_FILE_SIZE) {
+            return Promise.reject(
+              new FileBlobUploadError("file_too_large", file)
+            );
+          }
+
+          if (contentType.startsWith("image/")) {
```

Comment:
Why not creating a `isSupportedImageContentFragmentType` for consistency with the above? Also I guess we're not planning on supporting all image type, right?

Reply:
It's coming in the follow up PR as I need it widely in the BE.